### PR TITLE
[MinGW] Fix GCC ar detection.

### LINKED ISSUE #96736: `ar` version check in `platform/windows/detect.py` can fail if there is a patch number
### Tested versions

- Reproducible in v4.4.dev.custom_build [5675c7646] and likely any commit since https://github.com/godotengine/godot/commit/35a8ff9284edc2f942b7dca942f2a3c4f3698b92 as long as the version of `ar` does have a patch version

### System information

Windows 10, compiling with mingw-64, python version 3.11.9 (in msys2)

### Issue description

Short version: `detect.py` has a regex that finds the version of `ar` but it has a lookahead for the patch version, which counts as a group and the code later uses the lookahead group instead of the detected digit as the patch version, resulting in an error when comparing the version later.

Long version: When attempting to compile, there was an error that leads to line 674 of `detect.py` (https://github.com/godotengine/godot/commit/35a8ff9284edc2f942b7dca942f2a3c4f3698b92#diff-2308ebd6c211bc5c868068f7009dca599fcc991b99063393fce0f48d6b172d44R674). On that line, the regex group with offset 3 is used but that results in getting the patch number with the preceding period (so `.1` instead of `1`) which throws an error later when it tries to compare version numbers.

The commit that introduced this was made on the same day that the `binutils` package was updated in the msys2 repository: (https://github.com/msys2/MINGW-packages/commit/afbe0f87f45c51e927ba12d33d3c5afd3c9fe81e) so this was likely missed because the version prior to this update was 2.43, lacking a patch number and now it's 2.43.1, so a recently updated mingw installation will fail but an older one might not.

I fixed this for myself by just editing the same line in `detect.py` to the capture group with offset 4 instead. Can also change the offset in the conditional to 4 but it should work either way. Compilation completed properly several times, the binary runs.

### Steps to reproduce

Attempt to build the `master` branch on windows with mingw-64 with a `binutils` package that has a patch number in its version.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/96736

### CODE REVIEW

File: platform/windows/detect.py [hunk 1]
```diff
@@ -666,12 +666,12 @@ def get_ar_version(env):
         print_warning("Couldn't check version of `ar`.")
         return ret
 
-    match = re.search(r"GNU ar \(GNU Binutils\) (\d+)\.(\d+)(:?\.(\d+))?", output)
+    match = re.search(r"GNU ar \(GNU Binutils\) (\d+)\.(\d+)(:?\.)?(:?\d+)?", output)
```
Comment (akien-mga) [inline platform/windows/detect.py:669]: This looks weird to me. I feel this was meant to be a non capturing group, but the syntax is wrong, it should be `(?:\.`, not `(:?\.`.
```suggestion
    match = re.search(r"GNU ar \(GNU Binutils\) (\d+)\.(\d+)(?:\.(\d+))?", output)
```

**Review by akien-mga — APPROVED** [review]

File: platform/windows/detect.py (hunk 1 above)
Comment (mechalynx) [reply platform/windows/detect.py]: I guess AI can't parse regular languages when it's trained in a natural one (😏 )

**Comment by alvinhochun** [comment]
I thought I tested it, but apparently not... (facepalm)
Thanks for fixing.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**